lib/dnsimple/client/vanity_name_servers.rb

```diff
@@ -0,0 +1,43 @@
+module Dnsimple
+  class Client
+    module VanityNameServers
+
+      # Enable vanity name servers for a domain. 
+      #
+      # @see https://developer.dnsimple.com/v1/nameservers/vanity-nameservers/#enable
+      #
+      # @param  [#to_s] domain The domain id or domain name.
+      # @param  [Hash] names A hash of up to 4 external name servers; hash keys
+      # are ns1 through ns4, e.g.
+      #         {
+      #           "ns1" => "ns1.example.com",
+      #           "ns2" => "ns2.example.com"
+      #         }
+      #
+      # @return [void]
+      # @raise  [RequestError] When the request fails.
+      def enable_vanity_name_servers(domain, names)
```

Comment:
Technically, as of today we allow blank NS and we skip the blanks on our side. I would not be too much worried about validation here.

Just make sure that the Hash you create inside the method has higher priority over any value provided in `options`.